app/lib/search/updater.dart

```diff
@@ -159,20 +166,19 @@ class IndexUpdater implements TaskRunner {
     } on MissingAnalysisException catch (_) {
       // Nothing to do yet, keeping old version if it exists.
     }
-    await _updateSnapshotIfNeeded();
   }
 
   Future<void> _updateSnapshotIfNeeded() async {
-    final DateTime now = DateTime.now();
-    if (now.difference(_lastSnapshotWrite).inHours > 12) {
-      _lastSnapshotWrite = now;
-      try {
+    try {
+      if (await snapshotStorage.wasUpdatedRecently()) {
+        _logger.info('Snapshot update skipped (found recent snapshot).');
+      } else {
         _logger.info('Updating search snapshot...');
         await snapshotStorage.store(_snapshot);
         _logger.info('Search snapshot update completed.');
-      } catch (e, st) {
-        _logger.warning('Unable to update search snapshot.', e, st);
       }
+    } catch (e, st) {
```

Comment:
Nit: Let's put the catch-all inside the else-branch to make it as narrow as possible.

Reply:
I'd like to keep it there for now, otherwise there is a small chance that the storage bucket returns 5xx on the recency check, and it crashes the search instance. I'm adding a TODO to make it narrower.